js/scripts.js

```diff
@@ -1,17 +1,12 @@
-function changeColor(){
-    var hex_numbers = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9",
-    "A", "B", "C", "D", "E", "F"];
+function changeColor() {
+    var hex_numbers = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9","A", "B", "C", "D", "E", "F"];
 
     var hexcode = '';
 
-    for(var i=-0;i < 6;i++){
```

Comment:
This was the crux of the problem. You were generating a 7 character hex value, but hex values can only be 6 characters long. So when you tried looping 7 times, the color value you generated was too long and the DOM rejected the change.